### PR TITLE
[feat] create reset password flow

### LINKED ISSUE #91: Create reset password flow
https://linear.app/trap-garden/issue/TG-54/create-reset-password-flow

### PR DESCRIPTION
**Pull request by rachaelch3n**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- created reset your password page that checks for password complexity and if passwords match.
- follow exact same logic as signup page.
- send button only activates when user passes all complexity checks and password inputs are the same in both boxes.
- submission will check if password input is a new password. if not, error will appear and password won't be reset.
- if all criteria (complexity, confirm password, and new password) are all met, user is directed to login page where the new credentials should work.

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 
Screen Shot 2025-03-06 at 10 10 15 PM
Screen Shot 2025-03-06 at 10 10 33 PM
Screen Shot 2025-03-06 at 10 10 55 PM
Screen Shot 2025-03-08 at 1 34 00 AM

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Check for concision and error message logic (play around with typing in both text inputs). Also see if there's redundancy in the code.

Do we want the error message to appear while user is typing?

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

- what should text under the header be?

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @kylezryr

### CODE REVIEW

**Comment by kylezryr** [comment]
you'll need to rebase this branch now that the forgot password flow has been implemented! so:

1. git checkout main
2. git pull origin main
3. git checkout `branch name`
4. git rebase origin main

File: app/(auth)/email-sent/page.tsx
Comment (kylezryr) [inline app/(auth)/email-sent/page.tsx:1]: this page should be removed from the PR automatically after rebasing

File: app/(auth)/forgot-password/page.tsx
Comment (kylezryr) [inline app/(auth)/forgot-password/page.tsx:1]: this page should be removed from the PR automatically after rebasing

File: app/(auth)/forgot-password/reset-password/page.tsx
Comment (kylezryr) [inline app/(auth)/forgot-password/reset-password/page.tsx:1]: could we move this page to not be nested within `forgot-password` but to be its own folder in `(auth)`?

File: app/(auth)/forgot-password/reset-password/page.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      setIsSubmitted(true);
+      return;
+    }
+
+    try {
+      // Update the user's password
+      const { error } = await supabase.auth.updateUser({ password });
+
+      if (error) {
+        window.alert('Something went wrong. Please try again later.');
```
Comment (kylezryr) [inline app/(auth)/forgot-password/reset-password/page.tsx]: could we make this a `console.error` instead of `window.alert` with the error.message

File: app/(auth)/forgot-password/reset-password/page.tsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        throw new Error(
+          `An error occurred trying to sign out: ${signOutError.message}`,
+        );
+      }
+
+      // Redirect to confirm reset page
+      router.push('/login');
+    } catch (err) {
+      window.alert(
+        err instanceof Error ? err.message : 'Unexpected error occurred.',
```
Comment (kylezryr) [inline app/(auth)/forgot-password/reset-password/page.tsx]: can we also make this a `console.error` and log the err.message too

File: components/PasswordComplexity.tsx [hunk 3]
```diff
@@ -64,3 +64,20 @@ export function Requirement({ met, text }: { met: boolean; text: string }) {
     </P3>
   );
 }
+
+export function OnlyXRequirement({
```
Comment (kylezryr) [inline components/PasswordComplexity.tsx]: could we rename this to something more descriptive? maybe like `NewPasswordRequirement` or something

File: app/(auth)/styles.ts [hunk 4]
```diff
@@ -1,6 +1,40 @@
...
+
+export const GrayP3 = styled(P3)`
+  color: ${COLORS.midgray};
+`;
+
+export const ResetLinkButton = styled(StyledLinkButton)`
+  padding: 4px;
+`;
+
+export const ColumnFlexContainer = styled.div`
```
Comment (kylezryr) [inline app/(auth)/styles.ts]: you may not actually need this container, you can just use a `Flex` container imported from `styles/containers.ts` and assign the appropriate props inside of your tsx file

**Review by kylezryr — CHANGES_REQUESTED** [review]
looks good overall, not much to change functionality wise as it all works well, just a couple minor things to address!

**Review by kevin3656 — APPROVED** [review]
Looks good, great work Sashank !

File: app/reset-password/page.tsx [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+  };
+
+  const handleConfirmPasswordChange = (confirmPassword: string) => {
+    setVerifyNewPasswordError('');
+    setConfirmPassword(confirmPassword);
+  };
+
+  return (
+    <StyledForm>
+      <GreenH2>Set new password</GreenH2>
```
Comment (kylezryr) [inline app/reset-password/page.tsx]: note: in the future we will have text below this

**Review by kylezryr — APPROVED** [review]
looks good!
